src/wallet/wallet.go

```diff
@@ -648,5 +570,93 @@ func cmpUxOutByHash(a, b UxBalance) bool {
 		logger.Panic("Duplicate UxOut when sorting")
 	}
 	return cmp < 0
+}
+
+// ChooseSpends chooses uxouts from a list of uxouts.
+// It first chooses the uxout with the most number of coins that has nonzero coinhours.
+// It then chooses uxouts with zero coinhours, ordered by sortStrategy
+// It then chooses remaining uxouts with nonzero coinhours, ordered by sortStrategy
+func ChooseSpends(uxa []UxBalance, amt Balance, sortStrategy func([]UxBalance)) ([]UxBalance, error) {
```

Comment:
So whatever the strategy is, It always choose the uxout that has highest coins with none-hours first?

Reply:
Yes, because you need at least 1 input with coinhours.  So it looks at the one with the highest coins, since that is most likely to satisfy the send amount.

Then it tries to get the rest from inputs with no coinhours.  Keep in mind, most of the time they will have coinhours. Only someone spending frequently will have 0 coin hours, at least until we allow people to send outputs with 0 coinhours.

If it still needs more, then it gets from the remaining outputs with coinhours.